### DISCUSSION TITLE
MEMS I2S Audio for RPi Zero

### DISCUSSION THREAD

**Discussion by MichaelPWalter** [Nachtzuster/BirdNET-Pi#145] [Show and tell]
I wanted to run my BirdNET on a low cost, low power RPi Zero 2W (from here on as "Zero"). The Zero however lacks a good interface for audio input. In this post I wanted to showcase a cheap but robust way to add a microphone to the Zero without the need for adapters.
Using I2S, a digital audio interface standard, a MEMS microphone can be directly connected to the headers of the Zero. Theoretically MEMS microphones with the ADC right on the breakout board should also be less susceptible to interference. (I have not rigorously tested this yet).

Info: 
Raspberry Pi Zero 2W
Bookworm 64-bit
Mic is I2S MEMS Microphone Breakout - SPH0645 (https://www.adafruit.com/product/3421) or (https://www.aliexpress.us/item/2255800778086031.html)
I followed this assembly instruction: https://learn.adafruit.com/adafruit-i2s-mems-microphone-breakout/raspberry-pi-wiring-test

I initially ran into some issues. The Adafruit assembly instructions initially do not show a hookup for SEL but from the text it appears they mean to connect it to GND. After you hook up the six connections you power the Zero (wait a couple off minutes to let it boot) and ssh into it. 

Ill also attach a quick rundown of the software side:

- ssh into the Zero with your username and password. I used Putty on Windows.
- navigate to `/boot/firmware` (its three times `cd ..` followed by` cd /boot/firmware`)
- `sudo nano config.txt`
- towards the top of the file check that `dtparam=i2s=on` is uncommented
- on the bottom of the file add the line `dtoverlay=googlevoicehat-soundcard` from what I understand this enables the driver for the audio interface
- Close the file using CRTL+X and Y to confirm the override.
- `sudo reboot `to reboot the Zero
- ssh back into the Zero after a couple of min and check if the microphone is recognized using `arecord -l` thats a small L for list and not a 1
- you should see something like this 
`card 0: sndrpigooglevoi [snd_rpi_googlevoicehat_soundcar], device 0: Google voiceHAT SoundCard HiFi voicehat-hifi-0 [Google voiceHAT SoundCard HiFi voicehat-hifi-0]
  Subdevices: 0/1
  Subdevice #0: subdevice #0`

**Comment by wenestvedt** [comment]
I have a few GoogleAIY Voice cards (hats and bonnet) -- I should try this. Thanks!
